src/views/SynthView.vue

```diff
@@ -31,14 +59,45 @@ function updateEquaveShift(event: Event) {
     emit("update:equaveShift", value);
   }
 }
+
+onMounted(() => {
+  if (props.mainGain !== null) {
+    analyser.value = props.audioContext.createAnalyser();
+    props.mainGain.connect(analyser.value);
+    timeDomainVisualizer.value.initialize(analyser.value);
+  }
+});
+
+onUnmounted(() => {
+  if (props.mainGain !== null && analyser.value !== null) {
+    props.mainGain.disconnect(analyser.value);
+    analyser.value = null;
+  }
+});
 </script>
 
 <template>
   <main>
     <div class="columns-container">
       <div class="column synth-controls">
         <h1>Synth</h1>
-        <i>Coming soon to a browser near you...</i>
+        <div class="control-group">
+          <label for="volume">Main volume</label>
+          <input
+            type="range"
+            min="0"
+            max="0.5"
+            step="any"
+            v-model="mainVolume"
+          />
+        </div>
+        <h2>Waveform</h2>
+        <TimeDomainVisualizer
```

Comment:
This canvas element is ugly here. How do you even style those responsively? I'm pretty sure it has to be a canvas due to the amount of sound data it has to render.

Reply:
Used CSS width = 100% with offset width compensation while drawing the canvas.